sys/pm_callouts.c

```diff
@@ -408,6 +408,191 @@ void redir(portmaster_packet_info* packetInfo, portmaster_packet_info* redirInfo
     return;
 }
 
+void send_tcp_rst(portmaster_packet_info* packetInfo, void* originalPacket, ULONG originalPacketLength) {
+    // Only TCP is supported
+    if(packetInfo->protocol != 6) {
+        return;
+    }
+
+    if(packetInfo->ipV6) {
+        // Initialize header for the original packet with SYN flag
+        ULONG originalIPHeaderLength = calc_ipv6_header_size(originalPacket, originalPacketLength, NULL);
+        PIPV6_HEADER originalIPHeader = (PIPV6_HEADER) originalPacket;
+        PTCP_HEADER originalTCPHeader = (PTCP_HEADER) ((UINT8*)originalPacket + originalIPHeaderLength);
+
+        // Initialize variables
+        PNET_BUFFER_LIST injectNBL;
+        NTSTATUS status;
+        UINT16 packetLength = sizeof(IPV6_HEADER) + sizeof(TCP_HEADER);
+        void *tcpResetPacket;
+        PIPV6_HEADER ipHeader;
+        PTCP_HEADER tcpHeader;
+
+        // allocate memory for the reset packet
+        tcpResetPacket = portmaster_malloc(packetLength, FALSE);
+
+        // initialize IPv6 header
+        ipHeader = (PIPV6_HEADER) tcpResetPacket;
+        ipHeader->Version = 6;
+        ipHeader->Length = sizeof(TCP_HEADER);
+        ipHeader->NextHdr = 6;
```

Comment:
Use `packetInfo->protocol` insteads for more clarity. (Or, Add a comment to which protocol you are setting it. Yes it's obvious, but it's also not.)